src/fiatconnect/BankDetailsScreen.tsx

```diff
@@ -0,0 +1,275 @@
+      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
+    },
+    {
+      name: 'accountNumber',
+      label: i18n.t('fiatAccountSchema.accountNumber.label'),
+      regex: /^\d{10}$/,
+      placeholderText: i18n.t('fiatAccountSchema.accountNumber.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountNumber.errorMessage'),
+    },
+  ],
+}
+
+const BankDetailsScreen = ({ route, navigation }: Props) => {
+  const { t } = useTranslation()
+  const { providerURL, fiatAccountSchema } = route.params
+  const [validInputs, setValidInputs] = useState(false)
+  const [textValue, setTextValue] = useState('')
+  const [errors, setErrors] = useState(new Set())
+  const inputRefs = useRef<string[]>([textValue])
+
+  const dispatch = useDispatch()
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle: i18n.t('bankDetailsScreen.header'),
+    })
+  }, [navigation])
+
+  const getFieldsBySchema = (fiatAccountSchema: FiatAccountSchema): Fields[] => {
+    return SCHEMA_TO_FIELD_METADATA_MAP[fiatAccountSchema]
+  }
+
+  const formFields = useMemo(() => {
+    const fields = getFieldsBySchema(fiatAccountSchema)
+
+    for (let i = 0; i < fields.length; i++) {
+      inputRefs.current.push('')
+    }
+    return fields
+  }, [fiatAccountSchema])
+
+  const onPressNext = async () => {
+    validateInput()
+
+    if (validInputs) {
+      const body: { [k: string]: string } = {}
+      for (let i = 0; i < formFields.length; i++) {
+        body[formFields[i].name] = inputRefs.current[i]
+      }
+
+      await addNewFiatAccount(providerURL, fiatAccountSchema, body)
+        .then((data) => {
+          // TODO Tracking here
+          dispatch(showMessage(t('bankDetailsScreen.addFiatAccountSuccess')))
+        })
+        .catch((error) => {
+          // TODO Tracking here
+          if (error === FiatConnectError.ResourceExists) {
+            dispatch(showError(ErrorMessages.ADD_FIAT_ACCOUNT_RESOURCE_EXIST))
+          }
+          Logger.error(TAG, `Error adding fiat account: ${errors}`)
+        })
+
+      // TODO: navigate to the next screen
+    }
+  }
+
+  const onPressSelectPaymentOption = () => {
+    // TODO: tracking here
+    // todo: naviagte to select payment options screen
+  }
+
+  const validateInput = () => {
+    setValidInputs(false)
+    const newErrorSet = new Set(errors)
+
+    formFields.forEach((field, index) => {
+      const fieldVal = inputRefs.current[index].trim()
+
+      if (fieldVal && !field.regex.test(fieldVal)) {
+        newErrorSet.add(field.name)
+      } else if (newErrorSet.has(field.name)) {
+        newErrorSet.delete(field.name)
+      }
+    })
+
+    setErrors(newErrorSet)
+    if (newErrorSet.size > 0) {
+      return
+    } else {
+      setValidInputs(true)
+    }
+  }
+
+  const setInputValue = (value: string, index: number) => {
+    const inputs = inputRefs.current
+    inputs[index] = value
```

Comment:
```suggestion
    inputRefs.current[index] = value
```
code golf: inline variables